File: C#/SoftInvPersistance/DAO/PedidoDAO.cs
using softinv.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftInv.DAO
{
    interface PedidoDAO {
        int insertar(PedidoDTO pedido);
        PedidoDTO obtenerPorId(int id);
        int modificar(PedidoDTO pedido);
        int eliminar(PedidoDTO pedido);
        IList<PedidoDTO> listarPedidos(int id);
    }
}

File: backend JAVA/SoftInvPersistance/src/main/java/pe/edu/pucp/softinv/dao/PedidoDAO.java
package pe.edu.pucp.softinv.dao;
import pe.edu.pucp.softinv.model.Pedido.PedidoDTO;

import java.util.ArrayList;

public interface PedidoDAO {
    Integer insertar(PedidoDTO pedido);
    PedidoDTO obtenerPorId(Integer idPedido);
    Integer modificar(PedidoDTO pedido);
    Integer eliminar(PedidoDTO pedido);
    ArrayList<PedidoDTO> listarPedidos(Integer idCliente);
    PedidoDTO obtenerCarritoPorCliente(Integer idCliente);
    ArrayList<PedidoDTO> listarTodoPedidos();
    ArrayList<PedidoDTO> listarTodoPedidosPaginado(Integer pagina);
    Integer contarPedidos();
}
